code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -158,6 +158,12 @@
 /obj/item/ammo_box/magazine/internal/shot/toy/crossbow
 	max_ammo = 5
 
+//crossbow
+/obj/item/ammo_box/magazine/internal/shot/crossbow
+	ammo_type = /obj/item/ammo_casing/caseless/crossbowbolt && /obj/item/ammo_casing/apfsdsbolt

```

Comment:
@duncathan that's not really constant folding.
it's just storing the result of the && operation, which is a TRUE/FALSE (1/0) value.

Constant folding is more:
"this operation always happens with the same operands and the answer is always 7, I'm just going to remove the operation and put a 7."